### PR TITLE
Add the documentation updates for k8s (specs) / EFS and NF Schema

### PR DESCRIPTION
**Pull request by abhi18av**
The PR adds 

- [x] Head and Service pod spec docs for K8s
- [x] A reference for EFS 
- [x] Add the reference doc for NF schema

This work has been done in coordination with @ferralej

### CODE REVIEW

File: docs/compute-envs/k8s.md [hunk 1]
```diff
@@ -98,6 +98,14 @@ Default: the same as *Storage mount path*.
 
 Default is the `default` service account in your Kubernetes cluster.
 
+**4.** The pod behavior within the cluster could be controlled by using the **Pod cleanup policy** option.
+
+**5.** To specify custom specs for the `head pod` (i.e. `nf-compute-env-**`) using the **Custom head pod specs** option.
```
Comment (pditommaso) [inline docs/compute-envs/k8s.md:103]: the example in the brackets `(i.e. nf-compute-env-**)` is a bit out of context. Have a look in new field hint

File: docs/compute-envs/k8s.md [hunk 2]
```diff
@@ -98,6 +98,14 @@ Default: the same as *Storage mount path*.
...
+
+**6.** To specify custom specs for the `service pods` (i.e. `nf-workflow-**`) using the **Custom service pod specs** option.
```
Comment (pditommaso) [inline docs/compute-envs/k8s.md:105]: same here

File: docs/compute-envs/k8s.md (hunk 1 above)
Comment (abhi18av) [reply docs/compute-envs/k8s.md]: I've updated the comment for this.

File: docs/compute-envs/k8s.md [hunk 3]
```diff
@@ -98,6 +98,14 @@ Default: the same as *Storage mount path*.
 
 Default is the `default` service account in your Kubernetes cluster.
 
+**4.** The pod behavior within the cluster could be controlled by using the **Pod cleanup policy** option.
+
+**5.** To provide a custom configuration for the pod running the Nextflow workflow you can use the **Custom head pod specs** option.
+
+**6.** To provide a custom configuration for the compute environment service pod you can use the **Custom service pod specs** option.
+
```
Comment (pditommaso) [inline docs/compute-envs/k8s.md:106]: As a reader I don't understand what's 5 and 6 are useful? The text in the tips add something more 

> 5. This field allow you to provide a custom configuration for the pod running the Nextflow workflow e.g. nodeSelector and affinity constraints. It should be a valid PodSpec YAML structure starting with `spec:`.

> 6. This field allow you to provide a custom configuration for the compute environment service pod e.g. nodeSelector and affinity constraints. It should be a valid PodSpec YAML structure starting with `spec:`.

Comment (abhi18av) [reply docs/compute-envs/k8s.md:106]: Oh, I assumed that people who are reading this part would have this background anyhow. No worries, I'll update it to what you've mentioned in the tips there.

Comment (abhi18av) [reply docs/compute-envs/k8s.md]: This has now been resolved.

**Comment by pditommaso** [comment]
ok, this is good, tho the new K8s config is not yet in prod. Let's wait we deploy the new version

**Review by pditommaso — APPROVED** [review]

**Pull request merged**